instrumentation/apache-httpclient-4.0/src/main/java/io/opentelemetry/javaagent/instrumentation/hypertrace/apachehttpclient/v4_0/HttpEntityInstrumentation.java

```diff
@@ -114,8 +114,13 @@ public static void enter(
       if (contentSize <= 0 || contentSize == Long.MAX_VALUE) {
         contentSize = ContentLengthUtils.DEFAULT;
       }
-      ByteArrayOutputStream byteArrayOutputStream =
-          BoundedByteArrayOutputStreamFactory.create((int) contentSize);
+
+      // TODO set charset
+      String encoding =
+          thizz.getContentEncoding() != null ? thizz.getContentEncoding().getValue() : "";
```

Comment:
what encoding are you planning to use for the case where it is null ?

Reply:
this is wrong, the encoding here is not charset. The default charset is `ISO-8859-1`

Reply:
It has been fixed in the next commit